### PR TITLE
Multiple rbd volume enhancements

### PR DESCRIPTION
**Pull request by rootfs**
- looking for rbd device path through /dev/rbd/pool/image requires kube host install `ceph` package that contains `ceph-rbdnamer` udev helper. This inconvenience can be eliminated by getting rbd device path from sysfs. This fix obtains device path by examining sysfs.
- if rbd image is not formatted, format it to the designated filesystem type
- README update and more instructions

@saad-ali @kubernetes/rh-storage

### CODE REVIEW

File: pkg/volume/rbd/rbd_util.go [hunk 1]
```diff
@@ -37,19 +38,51 @@ import (
 	"k8s.io/kubernetes/pkg/volume"
 )
 
+// search /sys/bus for rbd device that matches given pool and image
+func getDevFromImageAndPool(pool, image string) string {
+	// /sys/bus/rbd/devices/X/name and /sys/bus/rbd/devices/X/pool
+	sys_path := "/sys/bus/rbd/devices"
+	if dirs, err := ioutil.ReadDir(sys_path); err == nil {
+		for _, f := range dirs {
+			name := f.Name()
```
Comment (brendandburns) [inline pkg/volume/rbd/rbd_util.go]: Seems like this whole thing could be done more effectively with a regular expression.

File: pkg/volume/rbd/rbd_util.go [hunk 2]
```diff
@@ -37,19 +38,51 @@ import (
...
+				return devicePath
+			}
+		}
+	}
+	return ""
+}
+
 // stat a path, if not exists, retry maxRetries times
-func waitForPathToExist(devicePath string, maxRetries int) bool {
+func waitForPath(pool, image string, maxRetries int) string {
```
Comment (brendandburns) [inline pkg/volume/rbd/rbd_util.go]: multi-value return? `(path string, found bool)`

File: pkg/volume/rbd/rbd_util.go (hunk 1 above)
Comment (rootfs) [inline pkg/volume/rbd/rbd_util.go]: that's cool, is there an efficient way to regex match two string patterns in two filename patterns?

File: pkg/volume/rbd/rbd_util.go (hunk 2 above)
Comment (rootfs) [inline pkg/volume/rbd/rbd_util.go]: that'll do, will refactor it

File: pkg/volume/rbd/rbd_util.go (hunk 1 above)
Comment (brendandburns) [inline pkg/volume/rbd/rbd_util.go]: Yeah you can use `|` to separate multiple regexps (e.g. `a.*b | b.*a`)

See:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/custom_column_printer.go#L41

for an example

Comment (brendandburns) [inline pkg/volume/rbd/rbd_util.go]: and you'll want to use matching groups `(...)` to select out the substring you're looking for.

Comment (rootfs) [inline pkg/volume/rbd/rbd_util.go]: thanks, updated, can you check again?

**Comment by rootfs** [comment]
@brendandburns PTAL thank you!

**Comment by brendandburns** [comment]
LGTM.  I kicked shippable

**Pull request merged**